File: domain/src/main/java/com/ricamgar/notify/domain/reminder/repository/RemindersRepository.java
package com.ricamgar.notify.domain.reminder.repository;

import com.ricamgar.notify.domain.reminder.model.Reminder;

import java.util.List;

import rx.Observable;

public interface RemindersRepository {

    Observable<List<Reminder>> getAll();

    Observable<Reminder> getById(long id);

    Observable<Reminder> addOrUpdate(Reminder reminder);

    Observable<Integer> delete(Reminder reminder);

    Observable<List<Reminder>> getTodoReminders();

    Observable<List<Reminder>> getHistoryReminders();
}

File: domain/src/main/java/com/ricamgar/notify/domain/reminder/repository/RemindersRepository.kt
package com.ricamgar.notify.domain.reminder.repository

import com.ricamgar.notify.domain.reminder.model.Reminder
import io.reactivex.Completable
import io.reactivex.Flowable

import io.reactivex.Observable
import io.reactivex.Single

interface RemindersRepository {

  val todoReminders: Flowable<List<Reminder>>

  val historyReminders: Flowable<List<Reminder>>

  fun getById(id: Long): Single<Reminder>

  fun addOrUpdate(reminder: Reminder): Single<Reminder>

  fun delete(id: Long): Completable
}
